tests/rules/assertions/genericSpacing.js

```diff
@@ -91,6 +90,68 @@ export default {
       options: ['always'],
       output: 'type X = Promise< (foo), bar, (((baz))) >',
     },
+
+    // Type annotations
+    {
+      code: 'const [state, setState] = useState<?string >(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState<?string > (null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState< ?string>(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState < ?string>(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState<? string>(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState< ? string>(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState< ? string >(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState < ? string > (null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'const [state, setState] = useState < ? string > ()',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'const [state, setState] = useState<?string>(null)',
+    },
+    {
+      code: 'useState<string >(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'useState<string>(null)',
+    },
+    {
+      code: 'useState< string>(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'useState<string>(null)',
+    },
+    {
+      code: 'useState< string >(null)',
+      errors: [{ message: 'There must be no space at start of type annotations' }],
+      output: 'useState<string>(null)',
+    },
```

Comment:
Could you add a test that's something like the following if it doesn't already exist elsewhere I'm interested to know if it strips all spaces blindly or only when it's between maybe types `?` and arrow brackets `</>`
```
    {
      code: 'useState< string | number >(null)',
      errors: [{ message: 'There must be no space at start of type annotations' }],
      output: 'useState<string | number>(null)',
    },
```